pynamodb/models.py

```diff
@@ -551,7 +551,10 @@ def from_raw_data(cls: Type[_T], data: Dict[str, Any]) -> _T:
             attr_name = cls._dynamo_to_python_attr(name)
             attr = cls.get_attributes().get(attr_name, None)  # type: ignore
             if attr:
-                attributes[attr_name] = attr.deserialize(attr.get_value(value))  # type: ignore
+                try:
+                    attributes[attr_name] = attr.deserialize(attr.get_value(value))  # type: ignore
+                except TypeError as e:
+                    raise ValueError(f'attribute: {attr_name}') from e
```

Comment:
I'm suggesting that we add a new exception type to `pynamodb/exceptions.py`, such as `AttributeDeserializationError` and then attach the attribute name as context there.

e.g

`raise AttributeDeserializationError(attr_name=attr_name) from e`

The `msg` can be constructed to include the attr name in the init of `AttributeDeserializationError`. I'd also consider subclassing `TypeError` for the new exception.

It would also be nice to add a test to cover this path